apps/docs-website/blog/2025-01-5-inversion-of-control.mdx

```diff
@@ -0,0 +1,162 @@
+
+```ts
+no-restricted-syntax: [
+    2,
+    ...baseNoRestrictedSyntaxRules,
+    // your custom rules here...
+],
+```
+
+While disabling some of the default ones is a little more involved.
+
+Every entry prints the index of itself at the end of the message property, so users should use the index to identify which entry they want to remove.
+
+In the docs we suggest using the native Javascript [`toSpliced()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced) method to remove an entry from the `baseNoRestrictedSyntaxRules` array:
+
+```ts
+no-restricted-syntax: [
+    2,
+    ...baseNoRestrictedSyntaxRules.toSpliced(2, 1)
+    // your custom rules here...
+],
+```
+
+[🠪 Learn more about this API in the docs](https://www.eslint-config-sheriff.dev/docs/configuration#customizing-the-no-restricted-syntax).
+
+### Going forward
+
+Recently an ESlint contributor released a very interesting alternative to the `no-restricted-syntax` rule called [eslint-no-restricted](https://github.com/bradzacher/eslint-no-restricted).
+
+Sheriff will most likely adopt this library in the future and abandon the current "homegrown" solution, as it seems to provide a superior DX.
+
+## Typescript-eslint `project` API
+
+Another case where Sheriff was swallowing up complexity at the cost of giving up a level of control to the user was the `parserOptions.project` API.
+
+### The problem
+
+Sheriff was passing the `tsconfig.json` to the ESLint config in a hard-coded way:
+
+```ts
+project: './tsconfig.json'
+```
+
+Naturally users with multiple `tsconfig.json` on the same level were feeling limited, because it wasn't possible to specify different paths for the `tsconfig.json`.
+
+### The solution
+
+The solution was to let the user define the `project` path through the Sheriff options:
+
+```ts
+pathsOverrides: {
+    tsconfigLocation: "./tsconfig.sample.json",
+},
+```
+
+and fallback to the [default one](https://typescript-eslint.io/blog/parser-options-project-true/) if not specified:
+
+```ts
+parserOptions: {
+    project: userChosenTSConfig || true,
+},
+```
+
+### Going forward
+
+Since then the typescript-eslint team moved to a new option called [`projectService`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta#project-service).
+
+This option is meant to definitely get rid of the need for the custom tsconfigs, like `tsconfig.eslint.json` etc, that some users had for some advanced use-cases. It also has other benefits. [Learn more](https://typescript-eslint.io/packages/parser#projectservice).
+
+Sheriff is currently exploring on adopting this new API ([Issue](https://github.com/AndreaPontrandolfo/sheriff/issues/203) | [PR](https://github.com/AndreaPontrandolfo/sheriff/pull/205)), but it seems like the feature is still experimental and [has some issues to iron out first](https://github.com/microsoft/vscode-eslint/issues/1911).
```

Comment:
```suggestion
Sheriff is currently exploring on adopting this new API ([issue](https://github.com/AndreaPontrandolfo/sheriff/issues/203) | [PR](https://github.com/AndreaPontrandolfo/sheriff/pull/205)).
```

The linked issue states that the project service is now more correct than the old version (in `typescript-eslint` `>=8.3.0`), although the issue remains open until issues (that also exist with the current version) are fixed. It's also no longer experimental in typescript-eslint v8, but has been stabilized.